### ISSUE TITLE
RocketChat crash with specially formatted message

### ISSUE THREAD

**Issue by dvessey-fes** [RocketChat/Rocket.Chat#30362]
### Description:

When sending a message with a specific format the rocketchat server process seems to hang

### Steps to reproduce:

1. Install latest rocketchat snap on ubuntu 20.04, do basic initial setup steps
2. Via web UI (currently using chrome 116.0.5845.187), start a DM with another fake user/yourself
3. Send the following in a message (no backticks):

```
|_ Urgency:  2 [14]
  |_ [5] asdf - asdf (Avg: 0:23:54 | Max: 1:05:42)
  |_ [4] asdf asdf  - asdf asdfa sdf  (Avg: 0:58:32 | Max: 2:05:22)
  |_ [3] asdfasdf sdf  safdf (Avg: 1:44:34 | Max: 1:44:55)
  |_ [1] asdfasdf asdft (Avg: 2:19:12 | Max: 2:19:12)
  |_ [1] asdf asdf asdf asdf (Avg: 1:51:37 | Max: 1:51:37)
|_ Urgency:  1 [12]
  |_ [7] asdfasdf a - Pasdfasdf  (Avg: 1:49:53 | Max: 3:17:14)
  |_ [1] asdfa sdf asdf  (Avg: 0:06:33 | Max: 0:06:33)
  |_ [1] asdf asdf - asdfsdf  (Avg: 0:35:55 | Max: 0:35:55)
  |_ [1] asdfasdfa sd (Avg: 1:38:25 | Max: 1:38:25)
  |_ [1] asdfasdf - asdfasdf a (Avg: 0:19:45 | Max: 0:19:45)
|_ Urgency:  0 [4]
  |_ [1] asdf asdf - asdf asdf a (Avg: 0:11:52 | Max: 0:11:52)
  |_ [1] asdfsdf - asdfasdf (ABC) asdf (Avg: 2:06:22 | Max: 2:06:22)
  |_ [1] asdfasdf  - asdfasdf asdf  (Avg: 0:21:15 | Max: 0:21:15)
  |_ [1] asdfasdf asd f (Avg: 1:02:28 | Max: 1:02:28)
|_ Urgency:  None [4]
  |_ [2] asdf asd f - asdfasdf asd f (Avg: 0:41:47 | Max: 0:43:43)
  |_ [1] asdfasd f - asdfasd f - asdfasdf (Avg: 0:21:22 | Max: 0:21:22)
  |_ [1] asdfasdf  - asdfasdf asd f (Avg: 0:03:52 | Max: 0:03:52)
```

### Expected behavior:

Message should just be posted.

### Actual behavior:

At this point the browser window seems to hang, and trying to open the URL in another tab gives a timeout/502

### Server Setup Information:

- Version of Rocket.Chat Server: 6.3.4
- Operating System: Ubuntu 20.04 (server)
- Deployment Method: snap
- Number of Running Instances: 1
- DB Replicaset Oplog: NA
- NodeJS Version: Whatever snap is
- MongoDB Version: Whatever snap is

### Client Setup Information

- Desktop App or Browser Version: chrome 116.0.5845.187
- Operating System: Windows 11

### Additional context

I've randomized the output but we have some reporting tools which spit out messages like this.

A partial fix seems to be to wrap the text in backticks. But this leads me to think that there's some issue with the server trying to parse the information?

Another interesting bit, if you only try to send a smaller snippet of the message it works, but it's delayed in posting/rendering.

Another note: My local vanilla test VM started out as 6.2.2 (hadn't booted it up in awhile), and this issue *didn't* happen there initially. But once it was updated to 6.3.4, then this issue was reproducible.

### Relevant logs:

I don't think these are needed, but let me know if you need them. The server side log output just stops after trying to send the message.

**Comment by nmagedman** [comment]
As an experiment, I tried removing elements, one at a time or in groups: `|`, `_`, `[1]`
It seems that it is the combination of the underscore with the square brackets.
My _guess_ is the combination of the two causes the Markdown parser to lock up, perhaps due to a pathological regexp?

**Comment by sujalcodes3** [comment]
Hello, I am new to Open Source. I would like to contribute to this issue. Can I get some help on where to start? @dvessey-fes

**Comment by Gummikavalier** [comment]
Still happens in RC 6.4.0

**Comment by Gummikavalier** [comment]
Still active in RC 6.4.2

**Comment by Gummikavalier** [comment]
RC 6.4.6 still active

**Comment by arpsyapathy** [comment]
RC 6.4.8 still active
RC [6.5.0-rc.8 still active
Developers, when will this be fixed?

**Comment by MartinSchoeler** [comment]
Hey, this PR fixes the issue https://github.com/RocketChat/fuselage/pull/1231
Expect the fix to come in version 6.5.0
Thanks!

**Comment by Gummikavalier** [comment]
Tested all three cases working now ok on RC 6.5.0.
